tests/Backend/Workspaces/MetadataFromRedshiftWorkspaceTest.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\Test\Backend\Workspaces;
+
+use Keboola\StorageApi\Workspaces;
+use Keboola\Test\Backend\Workspaces\WorkspacesTestCase;
+
+class MetadataFromRedshiftWorkspaceTest extends WorkspacesTestCase
+{
+    public function testCreateTableFromWorkspace()
+    {
+        // create workspace and source table in workspace
+        $workspaces = new Workspaces($this->_client);
+        $workspace = $workspaces->createWorkspace(["backend" => "redshift"]);
+        $connection = $workspace['connection'];
+        $db = $this->getDbConnection($connection);
+        $db->query("create table \"test.Languages3\" (
+                    \"Id\" varchar not null,
+                    \"Name\" varchar null
+                );");
+        // create table from workspace
+        $tableId = $this->_client->createTableAsyncDirect($this->getTestBucketId(self::STAGE_IN), array(
+            'name' => 'languages3',
+            'dataWorkspaceId' => $workspace['id'],
+            'dataTableName' => 'test.Languages3',
+        ));
+        $expectedIdMetadata = [
+            'KBC.datatype.type' => 'varchar',
+            'KBC.datatype.nullable' => '',
+            'KBC.datatype.basetype' => 'STRING',
+            'KBC.datatype.length' => '256',
+        ];
+        $expectedNameMetadata = [
+            'KBC.datatype.type' => 'varchar',
+            'KBC.datatype.nullable' => '1',
+            'KBC.datatype.basetype' => 'STRING',
+            'KBC.datatype.length' => '256',
+        ];
+        // check that the new table has the correct metadata
+        $table = $this->_client->getTable($tableId);
+        $this->assertEquals([], $table['metadata']);
+        $this->assertArrayHasKey('id', $table['columnMetadata']);
+        $this->assertMetadata($expectedIdMetadata, $table['columnMetadata']['id']);
+        $this->assertArrayHasKey('name', $table['columnMetadata']);
+        $this->assertMetadata($expectedNameMetadata, $table['columnMetadata']['name']);
+    }
+
+    public function testCopyImport()
```

Comment:
I'm getting this error inside mainQueuer worker:
```
message":"'4' is not valid length for int4
```

Reply:
I'm getting same error inside `mainQueue worker`.

I have switchted to the `piv-metadata-from-workspace` brach at the connection and do `composer install`. Is there something else I need to do ?

I have also rebase it localy on the master branche.